HW1/P3/P3.py

```diff
@@ -0,0 +1,25 @@
+import pyspark
+from pyspark import SparkContext
+sc = SparkContext("local[4]")
+import numpy as np
+sc.setLogLevel("ERROR")
+
+#import words and then split up the letters in each word and alphabetize them
+words = sc.textFile('words.txt')
+wordcharacters = words.map(lambda word: np.sort(list(word)))
+
+#now combine letters back together using join function. 
+SortedLetterSequence = wordcharacters.map(lambda x: "".join(x))
+
+#make two columns of our alphabetized words and .groupByKey to get anagrams
+keyvaluerdd = words.map(lambda x: ("".join(np.sort(list(x))),x))
+temp = keyvaluerdd.groupByKey().mapValues(list)
+Anagrams = temp.map(lambda x: (x[0],(len(x[1]),x[1])))
+
+#find maximum where we know max=11, so use filter.
+maximum = Anagrams.filter(lambda x: x[1][0] == 11)
```

Comment:
There should be a way to find the maximum (sorting it, for example) without a-priori knowing that it's going to have 11 occurrences. I think the 11 was just a hint to see if you were right or not, not to look up the max.